### PR TITLE
[9.x] BCC broken on 9.x with AWS SES Mail Transport

### PR DESCRIPTION
**Pull request by cuppett**
Redo of #41362
8.x version of same is #41389 

After upgrading to Laravel 9.x, bcc in our application stopped working
with the SES email transport. After reviewing the migration/upgrade
documentation and the code differences from 8.x -> 9.x, it's clear the
change in mailers likely exposed this; however, I also noticed a couple
fundamental, mechanical issues with the implementation:

1) Lack of Destinations specified in the array
2) No adherence to the 50 recipient limit of SES

I've corrected both of those and now can receive To/Cc/Bcc emails.

See Also: https://docs.aws.amazon.com/ses/latest/APIReference/API_SendRawEmail.html
See Also: https://docs.aws.amazon.com/aws-sdk-php/v2/api/class-Aws.Ses.SesClient.html#_sendRawEmail
See Also: https://docs.aws.amazon.com/ses/latest/dg/quotas.html

### CODE REVIEW

**Comment by taylorotwell** [comment]
This seems to attempt to address the recipient limit. How is BCC broken and how is it addressed in this PR?

**Comment by taylorotwell** [comment]
The code in 8.x did not seem to have this chunk or the `Destinations` key either though? So, how did this ever work on 8.x?

**Comment by cuppett** [comment]
> The code in 8.x did not seem to have this chunk or the `Destinations` key either though? So, how did this ever work on 8.x?

I didn't try 8.x in a debugger (just 9.x), but in 9.x I saw the $message->toString() report back:

```
From: somebody
To: somebody_else
// No BCC
Subject: Whatever

content
```

I bet if I did the same in 8.x, I'd see this:

```
From: somebody
To: somebody_else
Bcc: hidden_person
Subject: Whatever

content
```

And then the mail relay sniffs it out and strips that header when it's doing the relaying.

This difference in ->toString() is probably why Sendmail stopped working too (https://github.com/laravel/framework/issues/41357)

**Comment by cuppett** [comment]
(Ideally, we'd restore the old ->toString() output behavior; however, it's still dangerous to rely on it and continue to ignore the SES recipient limits)

File: src/Illuminate/Mail/Transport/SesTransport.php [hunk 1]
```diff
@@ -45,16 +51,21 @@ public function __construct(SesClient $ses, $options = [])
...
+            // See also: https://docs.aws.amazon.com/aws-sdk-php/v2/api/class-Aws.Ses.SesClient.html#_sendRawEmail.
+            foreach (collect($message->getEnvelope()->getRecipients())->chunk(self::RECIPIENT_LIMIT) as $recipients) {
+                $this->ses->sendRawEmail(
+                    array_merge(
+                        $this->options, [
+                            'Source' => $message->getEnvelope()->getSender()->toString(),
+                            'RawMessage' => [
+                                'Data' => $message->toString(),
+                            ],
+                            'Destinations' => $recipients,
```
Comment (taylorotwell) [inline src/Illuminate/Mail/Transport/SesTransport.php:64]: How is this array supposed to be formatted?

**Comment by taylorotwell** [comment]
Fixed here: https://github.com/laravel/framework/commit/8ca43f4c2a531ff9d28b86a7e366eef8adf8de84

**Pull request closed without merging**